Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -106,10 +106,12 @@ def fixed_name(file, bad_name)
       bad_name.sub(PREFIX_PLACEHOLDER, HOMEBREW_PREFIX)
     elsif bad_name.start_with? CELLAR_PLACEHOLDER
       bad_name.sub(CELLAR_PLACEHOLDER, HOMEBREW_CELLAR)
-    elsif (file.dylib? || file.mach_o_bundle?) && (file.parent + bad_name).exist?
+    elsif (file.dylib? || file.mach_o_bundle?) && (file.dirname/bad_name).exist?
       "@loader_path/#{bad_name}"
-    elsif file.mach_o_executable? && (lib + bad_name).exist?
+    elsif file.mach_o_executable? && (lib/bad_name).exist?
       "#{lib}/#{bad_name}"
+    elsif file.mach_o_executable? && (libexec/"lib"/bad_name).exist?
+      "#{libexec}/lib/#{bad_name}"
```

Comment:
Can we convert these to globs so it can support more flexible directory layouts like: `lib/foo/libfoo.dylib` or `lib/foo2.3/libfoo.dylib`

Reply:
I'd like to see some formulae where that's needed before we add that.

Reply:
Without it, I can't move all hashlink's libs to `/lib/hashlinlk` requiring `libhl.dylib` to stay symlinked to `/lib`. That clearly isn't an issue if I use `libexec` except I didn't wrote the final version of the formula that way and didn't get the impression that you want me to change it back to `libexec` 🤷

Reply:
Yeh, `libexec` seems better than `lib/hashlink`, apologies if I seemed to communicate otherwise.

Reply:
@MikeMcQuaid
```
==> Finishing up
ln -s ../Cellar/hashlink/1.11/bin/hl hl
Error: Failed to fix install linkage
The formula built, but you may encounter issues using it or linking other
formulae against it.
==> undefined local variable or method `libexec' for #<Keg:/usr/local/Cellar/hashlink/1.11>
/usr/local/Homebrew/Library/Homebrew/extend/os/mac/keg_relocate.rb:113:in `fixed_name'
```

Reply:
```diff
diff --git a/Library/Homebrew/keg_relocate.rb b/Library/Homebrew/keg_relocate.rb
index b8028ceab..bc9d32b87 100644
--- a/Library/Homebrew/keg_relocate.rb
+++ b/Library/Homebrew/keg_relocate.rb
@@ -117,6 +117,10 @@ class Keg
     path/"lib"
   end
 
+  def libexec
+    path/"libexec"
+  end
+
   def text_files
     text_files = []
     return text_files unless which("file") && which("xargs")

```
I guess this should do to the trick 🤔

Reply:
Opened #7683